src/client/components/pages/utils/filterByInstructorValues.ts

```diff
@@ -0,0 +1,32 @@
+import { MultiYearPlanResponseDTO } from 'common/dto/multiYearPlan/MultiYearPlanResponseDTO';
+
+/**
+ * Filters the array of multi-year plans by the faculty display name
+ */
+export const filterByInstructorValues = (
+  multiYearPlans: MultiYearPlanResponseDTO[],
+  instructorValues: Record<number, string>
+): MultiYearPlanResponseDTO[] => {
+  // Make a mutable ref to multiYearPlans to avoid parameter reassignment
+  let plans = multiYearPlans;
+  Object.entries(instructorValues)
+  // Skip falsy values or else we will filter out all rows with a blank instructor
+  // because of the .some() call, which returns false for an empty array.
+    .filter(([, value]) => !!value)
+    .forEach(([index, instructorFilter]) => {
+      const instructorFilterLower = instructorFilter.toLowerCase();
+      // Converting index to a number to appease compiler, which thinks index
+      // is a string, but we know that index will be a number.
+      const indexNum = parseInt(index, 10);
+      // Filter by the instructors
+      plans = plans
+        .filter((plan) => plan
+          .semesters[indexNum]
+          .instance.faculty.some((instructor) => instructor.displayName
```

Comment:
This line is throwing an error when I try to search within a column that does't have any data associated with it (i.e., there's no corresponding semester in the database).

Reply:
Thanks for catching this! I believe that the check for the truthiness of `plan.semesters[indexNum]` that I added in commit b4ff693 will fix this.

Reply:
Yep, that did the trick!